k8s/migration/pkg/utils/bmprovisionerutils.go

```diff
@@ -54,25 +54,102 @@ func ConvertESXiToPCDHost(ctx context.Context,
 		return err
 	}
 
+	if len(resources) == 0 {
+		return errors.New("no resources available from BM provisioner")
+	}
+
 	hs, err := GetESXiSummary(ctx, scope.Client, scope.ESXIMigration.Spec.ESXiName, vmwarecreds)
 	if err != nil {
 		return errors.Wrap(err, "failed to get ESXi summary")
 	}
 
-	for i := 0; i < len(resources); i++ {
-		if resources[i].HardwareUuid == hs.Hardware.SystemInfo.Uuid {
-			ctxlog.Info("Found a matching resource", "resource", resources[i].HardwareUuid, "name", resources[i].Hostname, "serial", resources[i].Id)
-			err := ReclaimESXi(ctx, scope, bmProvider, resources[i].Id, hs.Hardware.SystemInfo.Uuid)
-			if err != nil {
-				scope.ESXIMigration.Status.Phase = vjailbreakv1alpha1.ESXIMigrationPhaseFailed
-				updateErr := scope.Client.Status().Update(ctx, scope.ESXIMigration)
-				if updateErr != nil {
-					return errors.Wrap(updateErr, "failed to update ESXi migration status")
+	// Validate ESXi host has required hardware information
+	if hs.Hardware.SystemInfo.Uuid == "" {
+		ctxlog.Info("Warning: ESXi host has no hardware UUID")
+	}
+
+	// First, try to match by hardware UUID
+	var matchedResourceID string
+	var matchedResourceIdx int = -1
+	
+	if hs.Hardware.SystemInfo.Uuid != "" {
+		for i := 0; i < len(resources); i++ {
+			if resources[i].HardwareUuid != "" && resources[i].HardwareUuid == hs.Hardware.SystemInfo.Uuid {
+				ctxlog.Info("Found a matching resource by hardware UUID", 
+					"hardwareUuid", resources[i].HardwareUuid, 
+					"name", resources[i].Hostname, 
+					"id", resources[i].Id)
+				matchedResourceID = resources[i].Id
+				matchedResourceIdx = i
+				break
+			}
+		}
+	}
+
+	// If no UUID match found, fall back to MAC address matching
+	if matchedResourceIdx == -1 {
+		ctxlog.Info("No hardware UUID match found, trying MAC address matching")
```

Comment:
Remove the check above for hwuuid=="" and enhance this message saying either HWUUID was nil or no match was found falling over to checking the mac addr match. And print the HWUUID value in the log.